src/core/error_handling.h

```diff
@@ -32,27 +32,46 @@ log_error(char* msg);
  */
 
 // clang-format off
+#ifdef DEBUG_F
```

Comment:
There are unrelated code changes here which should be removed. Otherwise yes, splitting that PR in smaller chunks in the right approach and this LGTM.

Reply:
See: "Check out error_handling.h and Makefile from var-to-let" this was intended to improve the error messages generated.